gusto/time_discretisation.py

```diff
@@ -234,6 +234,128 @@ def apply(self, x_out, x_in):
         pass
 
 
+class ImplicitMultistage(TimeDiscretisation):
+    """
+    A class for implementing general diagonally implicit multistage (Runge-Kutta)
+    methods based on its Butcher tableau.
+
+    A Butcher tableau is formed in the following way for a s-th order
+    diagonally implicit scheme:
+
+    c_0 | a_00 a_01  .    a_0s
```

Comment:
This looks really nice in the code. Unfortunately I suspect it will look rubbish in the documentation, which doesn't know that these need to be on new lines!

For the documentation to be created nicely, I would add a `\n` at the end of each line when you know you want a line break. For an example, see here: https://github.com/firedrakeproject/gusto/blob/main/gusto/equations.py#L579

You might also want to fix this for the `ExplicitMultistage` scheme (sorry!)

Reply:
I have now done this one, separated out the butcher matrix comments as requested.